fix: avoid errors when deleting a resource

a-cordier opened the pull request:
Add null checks to document transformers to avoid technical exceptions
when deleting a resource.

see https://github.com/gravitee-io/issues/issues/6583
<!-- UI placeholder -->
🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6583-fix-delete-command-errors/index.html)
_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.
<!-- UI placeholder end -->

a-cordier committed:
fix: avoid errors when deleting a resource

Add null checks to document transformers to avoid technical exceptions
when deleting a resource.

see https://github.com/gravitee-io/issues/issues/6583

ytvnr reviewed (commented):
to be curious, do we know why we have those errors ?
is it normal that in some cases the object does not have a referenceId ?

a-cordier commented:
@ytvnr I honestly have no idea but am taking a closer look

ThibaudAV reviewed (approved):
Maybe a test should be added? ( I don't know what others  @gravitee-io/apim  think
 ) 

I did the following test : Added a Documentation md element then delete it.
With and without the patch and I don't have the error message in the console

So LGTM

a-cordier committed:
refactor: avoid duplicated deletion

Remove localy flag from the delete method of the SearchEngineService
as it is allways set to false.

Avoids calling deleteLocaly twice on the same document

a-cordier commented:
@ytvnr Added a commit with some refactor

The null checks are still needed as when the scheduler handles a delete command the only field it knows about is the document ID

@ThibaudAV added some tests

ytvnr reviewed (approved):
code seems good to me
